src/app/util/attribute-storage.cpp

```diff
@@ -1439,12 +1446,7 @@ bool IsFlatCompositionForEndpoint(EndpointId endpoint)
 
 bool IsTreeCompositionForEndpoint(EndpointId endpoint)
 {
-    uint16_t index = emberAfIndexFromEndpoint(endpoint);
-    if (index == kEmberInvalidEndpointIndex)
-    {
-        return false;
-    }
-    return emAfEndpoints[index].bitmask.Has(EmberAfEndpointOptions::isTreeComposition);
+    return !IsFlatCompositionForEndpoint(endpoint);
```

Comment:
This will trigger tree composition for unknown endpoint IDs.  Is that on purpose?